### ISSUE TITLE
Suggestion For Frost-For-Facebook : Add the features PRIVACY (PASSCODE LOCK), CHOOSE DOWNLOAD FOLDER, "ENABLE TEXT SELECTING, and SAVE DATA

### ISSUE THREAD

**Issue by ahyar92** [AllanWang/Frost-for-Facebook#1393]
![1.png](https://cdn.steemitimages.com/DQmbMfsKUKmgh4oWBcymDF2jBPvvWiMTw4T26ZZLmn4qudE/1.png)

![2.png](https://cdn.steemitimages.com/DQmU7d1GsAY5Cphdp79tnHNjiQiE6YZgjCvWH4Y91seuVhH/2.png)

### Components

The components (Features) I want to suggest is:
1. Privacy (PassCode Lock)
2. Choose Download Folder
3. Enable Text Selecting
4. Save Data

![2.png](https://cdn.steemitimages.com/DQmU7d1GsAY5Cphdp79tnHNjiQiE6YZgjCvWH4Y91seuVhH/2.png)

### Proposal Description
[Frost](https://github.com/AllanWang/Frost-for-Facebook) is a third party application for opening facebook that is very useful. I just got to know this kind of application from Frost, and after using it I was very impressed and didn't want to leave this application.

Frost is great, users can use multiple accounts in this application and can switch from one account to another quickly. And the most impressive thing in my opinion is that users can download videos on Facebook easily, this in my opinion greatly adds value to this application. And also the most interesting, users can change the color of the theme as desired, this will certainly provide a new color in playing Facebook.

I admit that Frost is interesting enough for users. But in my opinion, developers still need to make improvements (Development) to make it more interesting. All features that already exist on Frost are good enough and can provide a pleasant user experience. But there are still some interesting important features that are not yet available at Frost. At this time, I would like to suggest some interesting features for the Frost application, so as to make Frost more attractive and able to improve user experience later.

#### 1. Privacy (PassCode Lock)
Facebook users certainly have privacy issues that others don't want to know. And usually privacy matters are located on the chat page, and to protect all privacy, of course users need security to keep others from opening their Facebook. 

One solution is to add the "PassCode Lock" feature to secure the application so that it cannot be accessed by others. With the security of "PassCode Lock", of course all the privacy of Facebook users will be maintained. And only people who know the password can open the Frost application

#### 2. Choose Download Folder
Frost provides a Download feature that is able to download videos and photos on Facebook. But the storage folder for the download is still not manageable. With the "Choose Download Folder" feature, users can choose a storage folder or can even create a special folder to store the download results from the Frost application.

**Note:**
For the concept of work, you can see on points (Mockups / Examples)

#### 3. Enable Text Selecting
By using the "Enable Text Selecting" feature the user can do Text Selecting in Frost. In the Frost application, users cannot yet make a text selection on posts on Facebook. And I think, this feature is very necessary and must be added. This feature is useful for users who want to copy text on Facebook posts.

**Note:**
For the concept of work, you can see on points (Mockups / Examples)

#### 4. Save Data
The "Save Data" feature is very useful for users who want to access Facebook but have limited internet data. The way this feature works is very simple, when this feature is activated, all images on Facebook will not be displayed. This serves to maintain the user's internet data so as not to be wasteful. But it is certainly not fun, but it is very effective just to chat with friends or girlfriend on Facebook without fear of your internet data being drained a lot.

![2.png](https://cdn.steemitimages.com/DQmU7d1GsAY5Cphdp79tnHNjiQiE6YZgjCvWH4Y91seuVhH/2.png)

### Mockups / Examples
Below I will show examples of locations or illustrations if the features I recommend are implemented by the Developer.

![3.png](https://cdn.steemitimages.com/DQmVPaVzsrBqJvafov6u7aHAKN9ERByJLZs9m9VehmVu4yN/3.png)

![4.png](https://cdn.steemitimages.com/DQmVvWMdPhVsEsm3rwxKyMtsctYEPFbvoDbop4CmcYiuWb8/4.png)

![6.png](https://cdn.steemitimages.com/DQmWBr2fJYxtY1iXqcxPCFLMFTGNVn4aps5CvmqxFTDTC28/6.png)

![5.png](https://cdn.steemitimages.com/DQmUr3gxmKK6FcHM5hKv6rEAp8CPSa5v5RDvaZCanUPTGgN/5.png)

![7.png](https://cdn.steemitimages.com/DQmZ2UScV1kso3CAeuxv8zrkfjDPof77vCkmaEF6zAStBeQ/7.png)

-----

![8.png](https://cdn.steemitimages.com/DQmYLxr1XpCLa4gbXN8DBowMzDiTMVH9Ua2d7SWKbKksu47/8.png)

![9.png](https://cdn.steemitimages.com/DQmVCeHjKwwGJoiiUW3TAXupbLKfCbUz3bvej5DZR8bozRH/9.png)

-----

![10.png](https://cdn.steemitimages.com/DQmYvzCjc5bxn8HCqzorXB5hXzPeLoss1KPLhjWLGzvynoD/10.png)

![12.png](https://cdn.steemitimages.com/DQmXCENWcUFN4Xz1iqPkyw4nasWewMcEzhHVWyhndoZS3bk/12.png)

----

![13.png](https://cdn.steemitimages.com/DQmSkLJEPCMeCNbwGpQuQex7TcyMJR7xCGDq2junrNMMBZC/13.png)

![14.png](https://cdn.steemitimages.com/DQmUV1CwWEoANwcseucuT5zfrCHquWmt5nw2KysSq1cSJ7N/14.png)

![2.png](https://cdn.steemitimages.com/DQmU7d1GsAY5Cphdp79tnHNjiQiE6YZgjCvWH4Y91seuVhH/2.png)

### Benefits
The benefits of the features above are as follows:
#### 1. Privacy (PassCode Lock)
With the Privacy feature (PassCode Lock) users can secure the Frost application, so that other people cannot access it. And with this feature, of course all the privacy that is on the user's Facebook will be maintained safely.

#### 2. Choose Download Folder
With the "Choose Download Folder" feature, users can manage the storage folder of downloaded videos or photos from Facebook at Frost as desired. Even users can also create special folders for storing downloaded results from the Frost application

#### 3. Enable Text Selecting
With the "Enable Text Selecting" feature, users can do Text Selecting on posts available on Facebook. This feature is useful when users want to easily copy text in Facebook posts.

#### 4. Save Data
The "Save data" feature works for users who want to access Facebook with limited internet data. Users can access Facebook without worrying internet data will be wasteful, this feature is very effective when they want to chatting with friends on Facebook for a long time but the internet data is not wasteful.

![2.png](https://cdn.steemitimages.com/DQmU7d1GsAY5Cphdp79tnHNjiQiE6YZgjCvWH4Y91seuVhH/2.png)

### Conclusion
To implement the features that I suggest, developers may need to make major improvements to the Frost application. 4 The feature that I recommend is not a feature that is too large, but I am sure it will have a good impact to improve user experience later.

Of the 4 features that I recommend, the 3 initial features that I recommend are the ones I prioritize the most. While for "Save Data" maybe only a few users like it, but it won't be bad if the "Save Data" feature is added because it's a very unique feature.

And I hope, developers can immediately implement my suggestion.

Thank You :)

**Comment by AllanWang** [comment]
Nice assets!

1. Passcodes have been something I've wanted for a long time. There's still some consideration as to how I want to do it, and whether or not I want to use a library. I've considered making my own for quite some time, though it will definitely take time. Given the nature of Frost, other breaking changes tend to happen quite often, which takes precedence over these features.

3. On that note, Frost is also a web wrapper, which affects things like text selection. Right now, you should be able to long press articles to select text. I believe granular text selection comes out of the box for webviews. You can try it in your messages, though it may be inconsistent.

4. I believe early on, I experimented with disabling image loading when connected to mobile data. One of the problems is detecting when a connection changes between wifi and data, and requiring a user to restart the app on a change would be annoying. If I can figure out when the connection switches, this would be more useful.

2. This is a relatively easy feature, and I may consider it in the future.

---

To summarize

* I want to add fingerprint unlock, though I may need to implement it myself. Most solutions don't match the style and don't have theme customization
* I will consider adding a download folder selection since it's easy. Part of this is blocked by my current dialog library version, which I need to update. It will result in a lot of changes
* The other two are contingent on performance. They are things that will take time.

Unfortunately, as I am just one person with many projects, I cannot guarantee that any of these will happen any time soon. New problems pop up, and designs are subject to change.
